presto-main/src/main/java/com/facebook/presto/execution/CreateTableTask.java

```diff
@@ -85,16 +85,79 @@ public String explain(CreateTable statement, List<Expression> parameters)
         checkArgument(!statement.getElements().isEmpty(), "no columns for table");
 
         QualifiedObjectName tableName = createQualifiedObjectName(session, statement, statement.getName());
+
+        LinkedHashMap<String, ColumnMetadata> columns = new LinkedHashMap<>();
+        Map<String, Object> inheritedProperties = new HashMap<>();
+        prepareColumnMetadata(statement, metadata, session, tableName, columns, inheritedProperties);
+
         Optional<TableHandle> tableHandle = metadata.getTableHandle(session, tableName);
         if (tableHandle.isPresent()) {
+            TableMetadata tableMetadata = metadata.getTableMetadata(session, tableHandle.get());
             if (!statement.isNotExists()) {
-                throw new SemanticException(TABLE_ALREADY_EXISTS, statement, "Table '%s' already exists", tableName);
+                if (columnsEquals(columns.values(), tableMetadata.getColumns())) {
```

Comment:
The semantics for `CREATE TABLE IF NOT EXISTS` is that it ignores the error if the table already exists. There are no rules about it having the same schema. Connectors are free to impose their own restrictions, but these are connector specific and shouldn't be part of the engine.